docs/vcl.md

```diff
@@ -0,0 +1,80 @@
+## Introduction and Overview
+
+VCL is a communication library in VPP that provides a socket-based interface for client/server communication with VPP. It is designed to enable client applications to establish connections with VPP, send and receive control messages, and perform configuration and management operations.
+
+## VCL feature in CalicoVPP
+
+### Enabling VCL feature in CalicoVPP
+
+To enable vcl in your calicoVPP cluster, make sure parameter is set here:
+
+```yaml
+# dedicated configmap for VPP settings
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: calico-vpp-config
+  namespace: calico-vpp-dataplane
+data:
+  CALICOVPP_FEATURE_GATES: |-
+  {
+    "vclEnabled": true
+  }
+```
+
+### Using VCL in pods:
+
+To enable VCL on a pod, use the following annotation:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: samplepod
+  annotations:
+    "cni.projectcalico.org/vppVcl": "enable"
+```
+This will trigger vpp to create the VCL socket and add a session, which is an abstraction that allows applications to send and receive data over the network.
```

Comment:
This will only ask VPP to create the abstract socket, the session creation happens when you issue listens or connects calls over the VCL.
When we start the iperf container that's done by the iperf3 entrypoint that issues a listen over LDP+VCL